src/ucp/api/ucp_def.h

```diff
@@ -13,6 +14,8 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#define UCP_AM_ID_BITS 15
```

Comment:
let's remove this constant, instead the array of callbacks can grow on-demand with "realloc"